### PR TITLE
DSNS-116: Introducing a new dag airflow process for downloading of landsat level 1 data

### PR DESCRIPTION
**Pull request by gordonyeong**
Introducing a new dag airflow process that does the downloading of landsat level 1 data
in preparation of subsetting an similar and existing process in the usgsdownloader repo which is run by cron

### CODE REVIEW

File: dags/nci_ard/nci_ls_l1_downloader.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.providers.ssh.operators.ssh import SSHOperator
+from airflow.operators.dummy import DummyOperator
+
+pipeline_name = "L8L1C2"
+params = {
+    "project": "v10",
+    "walltime": "10:00:00",
```
Comment (dunkgray) [inline dags/nci_ard/nci_ls_l1_downloader.py:17]: these would be un-used. remove

**Review by dunkgray — COMMENTED** [review]
In general, let's clean up the params more:
in the params = {}
remove the keys used for qsub'ing
remove the airflow specific keys - e.g. cmd_timeout - no need to template them

move the params defined in the is_production_run code block (lines 55-60) in the  params = {}
section 

"/g/data/v10/eoancillarydata-2/" + params["pipeline_name"] + "_logs/" to
"/g/data/v10/work/landsat_downloads$" + params["pipeline_name"] + "_logs/"

File: dags/nci_ard/nci_ls_l1_downloader.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    # Production runs
+    # schedule_interval = "30 0 * * *"
+
+    schedule_interval = None
+
+    params["environment_var"] = "NCI-DEF-PROD "
+    params["umask"] = " umask 0017 "
+    params["logs_dir"] = (
+        "/g/data/v10/eoancillarydata-2/" + params["pipeline_name"] + "_logs/"
+    )
```
Comment (gordonyeong) [inline dags/nci_ard/nci_ls_l1_downloader.py:57]: @dunkgray did you mean to move these into the params?

If so, then it does convey that production parameters are made default. That is dangerous especially the    params["environment_var"] = "NCI-DEF-PROD " line.

File: dags/nci_ard/nci_ls_l1_downloader.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    # Production runs
+    # schedule_interval = "30 0 * * *"
+
+    schedule_interval = None
+
+    params["environment_var"] = "NCI-DEF-PROD "
+    params["umask"] = " umask 0017 "
+    params["logs_dir"] = (
+        "/g/data/v10/eoancillarydata-2/" + params["pipeline_name"] + "_logs/"
```
Comment (gordonyeong) [inline dags/nci_ard/nci_ls_l1_downloader.py:50]: @dunkgray  , are you happy for the logs to live here?

File: dags/nci_ard/nci_ls_l1_downloader.py (hunk 2 above)
Comment (dunkgray) [reply dags/nci_ard/nci_ls_l1_downloader.py:51]: yes.
It is getting the important variables into the param section.  Let's discuss @ 2

File: dags/nci_ard/nci_ls_l1_downloader.py (hunk 3 above)
Comment (dunkgray) [reply dags/nci_ard/nci_ls_l1_downloader.py:50]: I'm suggesting the logs can go here; /g/data/v10/work/landsat_downloads. /g/data/v10/work/ already has the ARD logs. Landsat download is not ancillary data.

File: dags/nci_ard/nci_ls_l1_downloader.py (hunk 2 above)
Comment (dunkgray) [reply dags/nci_ard/nci_ls_l1_downloader.py:51]: Another option is  "environment_var" = None in the params, so it is more visible as being part of the params, and keeping  params["environment_var"] = "NCI-DEF-PROD " where it is.

File: dags/nci_ard/nci_ls_l1_downloader.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    submit_task_id = "submit_downloader"
+    submit_downloader = SSHOperator(
+        task_id=submit_task_id,
+        command=COMMON
+        + """
+{{ params.umask }};
+
+mkdir -p {{ params.logs_dir }}{{ log_ext }};
+#!/bin/bash
```
Comment (dunkgray) [inline dags/nci_ard/nci_ls_l1_downloader.py:109]:
```suggestion

```

File: dags/nci_ard/nci_ls_l1_downloader.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+from os import environ
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.providers.ssh.operators.ssh import SSHOperator
+from airflow.operators.dummy import DummyOperator
+
+pipeline_name = "L8L1C2"
+params = {
+    "storage": "gdata/v10+scratch/v10+gdata/u46+scratch/u46",
```
Comment (dunkgray) [inline dags/nci_ard/nci_ls_l1_downloader.py:16]:
```suggestion
 
```

**Review by dunkgray — APPROVED** [review]
looks good - one minor suggestion

**Pull request merged**